### PR TITLE
dust off helm chart

### PR DESCRIPTION
**Pull request by Hermsi1337**
I've did some refactoring to the `helm`-chart by updating it to the latest chart-layout.

Furthermore I've updated the charts version, as well as the `appVersion`.

Unfortunately it still has to be `apiVersion: v1` since our charts-repository is yet not capable of handling `v2` charts. (see comment in `Chart.yaml`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ secrets and config maps available in multiple namespaces.
 
 ### Using Helm
 
-```shellsession
-$ helm upgrade --install kubernetes-replicator ./deploy/helm-chart/kubernetes-replicator
-```
+1. [Add the Mittwald-Charts Repo](https://github.com/mittwald/helm-charts/blob/master/README.md#usage)
```
Comment (martin-helmich) [inline README.md]: Maybe just so you don't have to switch pages to quickly setup the chart:

```suggestion
1. [Add the Mittwald-Charts Repo](https://github.com/mittwald/helm-charts/blob/master/README.md#usage):

    ```
    $ helm repo add mittwald https://helm.mittwald.de && helm repo update
    ```

File: deploy/helm-chart/kubernetes-replicator/templates/deployment.yaml [hunk 2]
```diff
@@ -3,40 +3,44 @@ kind: Deployment
...
   name: {{ include "kubernetes-replicator.fullname" . }}
   labels:
-    app.kubernetes.io/name: {{ include "kubernetes-replicator.name" . }}
-    helm.sh/chart: {{ include "kubernetes-replicator.chart" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- include "kubernetes-replicator.labels" . | nindent 4 }}
 spec:
-  replicas: 1
+  replicas: {{ .Values.replicaCount }}
```
Comment (martin-helmich) [inline deploy/helm-chart/kubernetes-replicator/templates/deployment.yaml]: This was previously hardcoded to `1` on purpose -- there's no locking/synchronization mechanism between instances of this controller, so running more than 1 of them would be quite racy.

File: deploy/helm-chart/kubernetes-replicator/values.yaml [hunk 3]
```diff
@@ -1,26 +1,37 @@
-# Default values for kubernetes-replicator.
-# This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
+replicaCount: 1
 
 image:
   repository: quay.io/mittwald/kubernetes-replicator
-  tag: stable
-  pullPolicy: IfNotPresent
+  #tag: stable # if no tag is given, the chart's appVersion is used
```
Comment (martin-helmich) [inline deploy/helm-chart/kubernetes-replicator/values.yaml]: If we don't set a tag anymore, don't we need to update the chart's `appVersion` with each deployment? Not to say that's a bad idea, but I didn't find the code for that anywhere. Or is that done by the mittwald/helm-charts repo workflow?

Comment (Hermsi1337) [reply deploy/helm-chart/kubernetes-replicator/values.yaml]: For now, the chart's `appVersion` has to be increased manually... but getting that done automagically on tags is a great idea. I'll give this a shot, gimme a sec :3

Comment (martin-helmich) [reply deploy/helm-chart/kubernetes-replicator/values.yaml]: >For now, the chart's appVersion has to be increased manually

You KNOW that you yourself would be the first one to forget this, dont't you? 😉

Comment (Hermsi1337) [reply deploy/helm-chart/kubernetes-replicator/values.yaml]: Hmm not sure... I haven't changed anything on the replicator itself, therefore there will be no new release. Instead I've updated the `version` of the chart.

**Pull request merged**